app/internal/usecase/llm_generate_usecase.go

```diff
@@ -70,39 +70,97 @@ func (u *llmGenerateUsecase) LLMGenerate(c echo.Context, req model.LLMGenerateRe
 	}
 
 	// 4. 質問ごとに回答を生成
-	answers := make([]model.LLMGeneratedResponse, 0, len(questions))
+	var wg sync.WaitGroup
+
+	type indexedResponse struct {
+		index int
+		resp  model.LLMGeneratedResponse
+	}
+	responseCh := make(chan indexedResponse, len(questions))
+	errorCh := make(chan error, len(questions))
 
 	llmModel := model.GeminiFlashLite
 	if model.LLMModel(req.Model) != "" {
 		llmModel = model.LLMModel(req.Model)
 	}
 
-	for _, question := range questions {
-		// プロンプトを作成
-		prompt := u.buildPrompt(question, companyInfo, &experience, req.Company)
+	for i, question := range questions {
+		wg.Add(1)
+		go func(idx int, q string) {
+			defer func() {
+				if r := recover(); r != nil {
+					errorCh <- fmt.Errorf("質問「%s」の処理中にパニックが発生: %v", q, r)
+					wg.Done()
+				}
+			}()
+
+			defer wg.Done()
```

Comment:
- 今のままの実装だとpanicした時に`wg.Done()`がたぶん2回呼び出される
- 単純にrecoverの`wg.Done()`を消したら、panicに入る→deferはスタックのため`wg.Done()`が先に実行される→recoverする→その瞬間`wg.Wait()`のブロッキングが無くなりerrorChが閉じる→既に閉じたerrorChに書き込んでしまう となってしまうかもしれない（全然そんなことはない可能性も大いにある、調べてくれてもいいよ😘）
- なので、recoverの中の`wg.Done()`は消して、`defer wg.Done()`を先に持ってきた方がいいと思います
```suggestion
			defer wg.Done()
			defer func() {
				if r := recover(); r != nil {
					errorCh <- fmt.Errorf("質問「%s」の処理中にパニックが発生: %v", q, r)
				}
			}()
```

Reply:
直したんだけどmainブランチの変更取り入れるところと一緒に修正をコミットしてしまってとても見づらいですごめんなさい（多分94行目とかですね）
[f99fee4](https://github.com/Teamsasa/es-writer-api/pull/24/commits/f99fee41611ea784eebdb572d1e1b769cefa7a7c)